enos/modules/vault_agent/main.tf

```diff
@@ -0,0 +1,69 @@
+terraform {
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+    }
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+variable "vault_agent_template_destination" {
+  type        = string
+  description = "The destination of the template rendered by Agent"
+}
+
+variable "vault_agent_template_contents" {
+  type        = string
+  description = "The template contents to be rendered by Agent"
+}
+
+variable "vault_root_token" {
+  type        = string
+  description = "The Vault root token"
+}
+
+variable "vault_instances" {
+  type = map(object({
+    private_ip = string
+    public_ip  = string
+  }))
+  description = "The Vault cluster instances that were created"
+}
+
+variable "vault_instance_count" {
+  type        = number
+  description = "How many vault instances are in the cluster"
+}
+
+variable "vault_install_dir" {
+  type        = string
+  description = "The directory where the Vault binary will be installed"
+}
+
+locals {
+  vault_instances = {
+    for idx in range(var.vault_instance_count) : idx => {
+      public_ip  = values(var.vault_instances)[idx].public_ip
+      private_ip = values(var.vault_instances)[idx].private_ip
+    }
+  }
+}
+
+resource "enos_remote_exec" "set_up_approle_auth_and_agent" {
+  content = templatefile("${path.module}/templates/set-up-approle-and-agent.sh", {
```

Comment:
Rather than using a `templatefile` you can use a normal script with environment variables. The remote exec resource supports an `environment` attribute which is a map. Using environment variables has the advantage that the script is testable without running Terraform since you can just set the env vars that you need and execute the script directly.

Reply:
Is there a particular reason when one is preferred over the other? I chose this as the examples I found in the enos directory seemed to use this approach, so I'd like to better understand when I should use one approach over the other.